src/tests/gov/nasa/jpf/jvm/FileVersionTest.java

```diff
@@ -0,0 +1,52 @@
+package gov.nasa.jpf.jvm;
+
+import gov.nasa.jpf.util.test.TestJPF;
+import gov.nasa.jpf.vm.ClassParseException;
+import org.junit.Test;
+
+import java.io.*;
+
+public class FileVersionTest extends TestJPF {
+
+    private static final String RESOURCES_PATH = "resources/";
+    private static final String JAVA17_CLASS = RESOURCES_PATH + "TestClassJava17.class";
+    private static final String JAVA21_CLASS = RESOURCES_PATH + "TestClassJava21.class";
+
+    // loading a .class file into a byte array
+    private byte[] loadClassFile(String resourceName) throws IOException {
+        try (InputStream is = getClass().getResourceAsStream(resourceName)) {
+            if (is == null) throw new IOException("Resource not found: " + resourceName);
+
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            // we choose here buffer size 1024 cause its enough to read most .class files in one or two iterations
+            // smaller buffer size like 256 will require more operations and larger buffers will waster memory
+            // i'm not sure which is suitable for this since the test classes we complied is empty but i think both could work
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            while ((bytesRead = is.read(buffer)) != -1) {
+                bos.write(buffer, 0, bytesRead);
+            }
+            return bos.toByteArray();
+        }
+    }
+
+    @Test
+    public void testSupportedVersionJava17() throws IOException {
+        byte[] classData = loadClassFile(JAVA17_CLASS);
+
+        ClassFile classFile = new ClassFile(classData);
+
+        assertTrue("Java 17 class file should be parsed successfully", true);
+    }
+
+    /*
+    @Test(expected = ClassParseException.class)
+    public void testUnsupportedVersionJava21() throws IOException {
+        byte[] classData = loadClassFile(JAVA21_CLASS);
+
+        ClassFile classFile = new ClassFile(classData);
```

Comment:
Please enable this test and ensure it throws an exception. The test looks good; what was the reason for commenting it out? Perhaps JPF loads the version of the class file that was compiled by the build process rather than the one in `src/tests`, which would explain that this test does not throw an exception and thus fails?
In that case, try to find a way to make your manually uploaded class file take precedence in the class loader.